.circleci/generate.sh

```diff
@@ -184,36 +194,43 @@ fi
 
 # add new or modified tests to the sets of tests to be repeated
 if (!($all)); then
-  add_diff_tests ()
-  {
-    dir="${BASEDIR}/../${2}"
-    diff=$(git --no-pager diff --name-only --diff-filter=AMR ${BASE_BRANCH}...HEAD ${dir})
-    tests=$( echo "$diff" \
-           | grep "Test\\.java" \
-           | sed -e "s/\\.java//" \
-           | sed -e "s,^${2},," \
-           | tr  '/' '.' \
-           | grep ${3} )
-    for test in $tests; do
-      echo "  $test"
-      has_env_vars=true
-      if echo "$env_vars" | grep -q "${1}="; then
-        env_vars=$(echo "$env_vars" | sed -e "s/${1}=/${1}=${test},/")
-      elif [ -z "$env_vars" ]; then
-        env_vars="${1}=${test}"
-      else
-        env_vars="$env_vars|${1}=${test}"
-      fi
-    done
-  }
+  # Sanity check that the referenced branch exists
+  if ! git show ${BASE_BRANCH} -- >&/dev/null; then
+    echo "Unknown base branch: ${BASE_BRANCH}. Skipping detection of changed tests."
+    echo "If you wish to detect changed tests, please use the '-b' flag or set the"
+    echo "BASE_BRANCH environment variable to an existing branch name (e.g. origin/${BASE_BRANCH})."
```

Comment:
We could use the `die` function that immediately returns with exit code 1, so we don't have to put the diff code in an `else` branch.

Reply:
Ah, the intention is continuing with the rest of the script, isn't it? I think I'd rather fail and demand the users to provide a base branch, so they can't skip the detection of changed tests. Otherwise, the warning message can be easily be overlooked.

Reply:
Perhaps the example, `(e.g. origin/${BASE_BRANCH}).`, could also include some other remote, as in `(e.g. origin/${BASE_BRANCH}, apache/${BASE_BRANCH}, etc.).`

Reply:
I don't really want it to `print_help` if the branch can't be found, since the message already explains proper usage. I can just `exit 2` here instead.